paywall/src/hooks/useOptimism.js

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useRef, useReducer } from 'react'
+
+import useLocksmith from './useLocksmith'
+import { OPTIMISM_POLLING_INTERVAL } from '../constants'
+
+export default function useOptimism(transaction) {
+  const API = transaction ? `/transaction/${transaction.hash}/odds` : ''
+  const hash = transaction ? transaction.hash : false
+  const status = transaction ? transaction.status : 'inactive'
```

Comment:
What is the `inactive` status?

Reply:
this is used to detect the state where the paywall is just sitting around, and the user has not clicked the "purchase" button yet. In that case, we have no transaction at all, but rather than do an awkward internal comparison of `(transaction ? transaction.status : 'inactive') === 'pending'` (for example), this pulls it up top so we can simply compare `status === 'pending'`